### PR TITLE
Fix issue - Joined geometry does not convert correctly

### PR DESCRIPTION
**Pull request by ZiyunShang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Fix issue - Joined geometry does not convert correctly.
https://github.com/DynamoDS/DynamoRevit/issues/2400 

This should be delayed to merge after Revit 2020 release.

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] Snapshot of UI changes, if any.

### Reviewers

@AndyDu1985 

### FYIs

@QilongTang @mjkkirschner

### CODE REVIEW

File: src/Libraries/RevitNodes/GeometryConversion/CurveLoopPartition.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+using BoundingBox = Autodesk.DesignScript.Geometry.BoundingBox;
+using PolyCurve = Autodesk.DesignScript.Geometry.PolyCurve;
+
+namespace Revit.GeometryConversion
+{
+
+    [SupressImportIntoVM]
+    internal static class CurveLoopPartition
+    {
+        public static List<List<List<Curve>>> ByCurveLoopsAndFace(Face face, List<List<Curve>> curveLoops)
```
Comment (AndyDu1985) [inline src/Libraries/RevitNodes/GeometryConversion/CurveLoopPartition.cs]: Is the parameter `face` used in this method?

File: src/Libraries/RevitNodes/GeometryConversion/CurveLoopPartition.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                {
+                    innerCurveLoops.Add(curveLoop);
+                }
+            }
+
+            var partitionedLoops = new List<List<List<Curve>>>();
+
+            // for each outer loop, collect the loops that are inside of it
+            // forming the partitions of the original curve loop set
+            foreach (var outerLoop in outerCurveLoops)
```
Comment (AndyDu1985) [inline src/Libraries/RevitNodes/GeometryConversion/CurveLoopPartition.cs]: Do you means that there will be several outer loops for one face?

File: src/Libraries/RevitNodes/GeometryConversion/CurveLoopPartition.cs [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                // remove innerEdge loops that have already been added to a partition
+                innerCurveLoops = innerCurveLoops.Where((_, j) => mask[j]).ToList();
+                // add the new partition
+                partitionedLoops.Add(comp);
+            }
+
+            return partitionedLoops;
+        }
+
+        public static List<List<Autodesk.Revit.DB.Curve>> GetAllCurveloopsFromRevitFace(Face face)
```
Comment (AndyDu1985) [inline src/Libraries/RevitNodes/GeometryConversion/CurveLoopPartition.cs]: Return nested lists makes the function unclear, I suggest that just return CurveLoop.

File: src/Libraries/RevitNodes/GeometryConversion/RevitToProtoFace.cs [hunk 4]
```diff
@@ -19,42 +19,42 @@ public static class RevitToProtoFace
         {
             if (revitFace == null) throw new ArgumentNullException("revitFace");
 
-            var revitEdgeLoops = EdgeLoopPartition.GetAllEdgeLoopsFromRevitFace(revitFace);
-            var partitionedRevitEdgeLoops = EdgeLoopPartition.ByEdgeLoopsAndFace(revitFace, revitEdgeLoops);
+            var revitCurveLoops = CurveLoopPartition.GetAllCurveloopsFromRevitFace(revitFace);
+            var partitionedRevitCurveLoops = CurveLoopPartition.ByCurveLoopsAndFace(revitFace, revitCurveLoops);
```
Comment (AndyDu1985) [inline src/Libraries/RevitNodes/GeometryConversion/RevitToProtoFace.cs]: Could we just use `var curveLoops = face.GetEdgesAsCurveLoops();` to get all curve loops of a face and distinguish the inner and outer? Then use them to create face? I think nested lists or nested nested lists is not a good code practice.

File: src/Libraries/RevitNodes/GeometryConversion/CurveLoopPartition.cs (hunk 2 above)
Comment (ZiyunShang) [reply src/Libraries/RevitNodes/GeometryConversion/CurveLoopPartition.cs]: Yes, sometimes there would be 2 or more outer loops.

File: src/Libraries/RevitNodes/GeometryConversion/CurveLoopPartition.cs (hunk 1 above)
Comment (ZiyunShang) [reply src/Libraries/RevitNodes/GeometryConversion/CurveLoopPartition.cs]: It looks like the face is no use in this method now, I will remove it.

**Pull request merged**